pkg/ride/tree_estimatorV4.go

```diff
@@ -96,130 +102,132 @@ func (e *treeEstimatorV4) wrapFunction(node *ast.FunctionDeclarationNode) ast.No
 	return block
 }
 
-func (e *treeEstimatorV4) walk(node ast.Node, enableInvocation bool) (int, error) {
+func (e *treeEstimatorV4) walk(node ast.Node) (int, bool, error) {
 	switch n := node.(type) {
 	case *ast.LongNode, *ast.BytesNode, *ast.BooleanNode, *ast.StringNode:
-		return 0, nil
+		return 0, false, nil
 
 	case *ast.ConditionalNode:
-		ce, err := e.walk(n.Condition, enableInvocation)
+		ce, ci, err := e.walk(n.Condition)
 		if err != nil {
-			return 0, errors.Wrap(err, "failed to estimate the condition of if")
+			return 0, false, errors.Wrap(err, "failed to estimate the condition of if")
 		}
 		cs := e.scope.save()
-		le, err := e.walk(n.TrueExpression, enableInvocation)
+		le, li, err := e.walk(n.TrueExpression)
 		if err != nil {
-			return 0, errors.Wrap(err, "failed to estimate the true branch of if")
+			return 0, false, errors.Wrap(err, "failed to estimate the true branch of if")
 		}
 		ls := e.scope.save()
 		e.scope.restore(cs)
-		re, err := e.walk(n.FalseExpression, enableInvocation)
+		re, ri, err := e.walk(n.FalseExpression)
 		if err != nil {
-			return 0, errors.Wrap(err, "failed to estimate the false branch of if")
+			return 0, false, errors.Wrap(err, "failed to estimate the false branch of if")
 		}
 		if le > re {
 			e.scope.restore(ls)
 			res, err := common.AddInt(ce, le)
 			if err != nil {
-				return 0, err
+				return 0, false, err
 			}
-			return res, nil
+			return res, ci || li, nil
 		}
 		res, err := common.AddInt(ce, re)
 		if err != nil {
-			return 0, err
+			return 0, false, err
 		}
-		return res, nil
+		return res, ci || ri, nil
```

Comment:
Would be also nice, if we add a comment above about `ci`, `ri`, `li`, `le`, `ce` explaining abbreviation

Reply:
Ok, I'll try, but they so local, so the short name is quite ok.